pkg/trustedresources/verify.go

```diff
@@ -155,14 +187,22 @@ func verifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.
 		// if this policy fails the verification and the mode is not "warn", should return error directly. No need to check other policies
 		if !passVerification {
 			if p.Spec.Mode == v1alpha1.ModeWarn {
+				failWarnPolicy = true
 				logger := logging.FromContext(ctx)
-				logger.Warnf("%w: resource %s in namespace %s fails verification", ErrResourceVerificationFailed, resource.GetName(), resource.GetNamespace())
+				msg := fmt.Sprintf("%v: resource %s in namespace %s fails verification", ErrResourceVerificationFailed, resource.GetName(), resource.GetNamespace())
```

Comment:
Is there a more helpful error message we could return that explains the reason why verification failed so that the user could fix it?

Reply:
How would the user be expected to resolve the error though? It could have failed due to no matching verification policies, a mismatched signature, or somehting else.

Reply:
"no matching verification policies" will be another error and returned before this error. 
This error maybe 1) bad signature 2) resource modified 3) bad public key(e.g. key is not updated) 
But we cannot tell which is the reason from our side. 😞

Reply:
ah ok-- good to know! I think it's totally reasonable to not separate out types of signature verification issues especially if we can't tell the difference.